### PR TITLE
ggml-cuda : explicitly use cmath for log2

### PR DESCRIPTION
**Pull request by slaren**
Should fix compilation issues when building with some versions of the standard library, as reported in #3398

### CODE REVIEW

File: ggml-cuda.cu [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #include <algorithm>
+#include <assert.h>
```
Comment (Cebtenzzre) [inline ggml-cuda.cu:2]: Shouldn't this be `#include <cassert>`?

Comment (slaren) [reply ggml-cuda.cu:2]: Yes, and it's not the only one, but all I did here is sort the includes alphabetically.

**Review by ggerganov — APPROVED** [review]

**Comment by ggerganov** [comment]
Should we merge this?

**Comment by slaren** [comment]
The person who reported the issue never responded, and nobody reported this issue again, so I think we can close this.

**Pull request closed without merging**